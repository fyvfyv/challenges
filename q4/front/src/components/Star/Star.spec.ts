import { mount } from '@vue/test-utils'
import Component from './index'

describe('Star', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Component, {
      propsData: {
        isFilled: false
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders grey colored correctly', () => {
    const wrapper = mount(Component, {
      propsData: {
        isFilled: false
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders yellow colored correctly', () => {
    const wrapper = mount(Component, {
      propsData: {
        isFilled: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
