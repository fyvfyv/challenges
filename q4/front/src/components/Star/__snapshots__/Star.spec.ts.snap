// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star renders grey colored correctly 1`] = `
<svg
  class=""
  height="25"
  width="23"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style="fill-rule: nonzero;"
  />
</svg>
`;

exports[`Star renders yellow colored correctly 1`] = `
<svg
  class="filled"
  height="25"
  width="23"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style="fill-rule: nonzero;"
  />
</svg>
`;
